aws/data_source_aws_nat_gateway.go

```diff
@@ -0,0 +1,114 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsNatGateway() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsNatGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"filter": ec2CustomFiltersSchema(),
```

Comment:
Can you also add the tags part, following @ewbankkit 's [comment](https://github.com/terraform-providers/terraform-provider-aws/pull/1294#issuecomment-331744990)?